File: src/app/shared/models/rule.ts
export interface RulesContainer {
  title: RulesCategory;
  prologue?: string;
  rules: Rule[];
  epilogue?: string;
}

export interface Rule {
  id: number;
  number?: string;
  title?: string;
  description: string;
}

export class RulesCategory {
  static readonly Principali = "Principali";
  static readonly Extra = "Extra";
  static readonly Penalità = "Penalità";
  static readonly Sacrificio = "Sacrificio";
  static readonly Bevuta = "La Bevuta";
}

File: nazdrovia/lib/models/classes/rule.dart
import 'package:json_annotation/json_annotation.dart';

part 'rule.g.dart';

@JsonSerializable()
class Rule {
  final int id;
  final String number;
  final String title;
  final String description;

  const Rule(this.id, this.number, this.title, this.description);

  factory Rule.fromJson(Map<String, dynamic> json) => _$RuleFromJson(json);
  Map<String, dynamic> toJson() => _$RuleToJson(this);
}

abstract class RulesCategory {
  static const PRINCIPALI = "Principali";
  static const EXTRA = "Extra";
  static const PENALITA = "Penalità";
  static const SACRIFICIO = "Sacrificio";
  static const BEVUTA = "La Bevuta";
}
